### ISSUE TITLE
r/aws_autoscaling_group: Remove deprecated `tags` argument

### ISSUE THREAD

**Issue by anGie44** [hashicorp/terraform-provider-aws#22905]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `tags` argument was deprecated in the v4.0 major milestone and should be removed in the next one (v5.0).

Relates:

* https://github.com/hashicorp/terraform-provider-aws/issues/7655
* https://github.com/hashicorp/terraform-provider-aws/pull/22663

**Comment by postmaxin** [comment]
Hi,
   I am just seeing this warning attempting to upgrade to 1.1. What is the replacement for "tags"? "tag" doesn't seem sufficient. For example, I have a few modules that accept a list of tags as an input and apply all of those tags to all resources within the module. These are merged with standard tags for that module, and then, in turn, applied to resources:

```
  tags = concat(local.asg_standard_tags, [
    {
      key = "Name"
      value = "${local.azs[count.index]}.disk.${module.k8s_common.cluster_fqdn}"
      propagate_at_launch = true
    },
    {
      key = "chef_bootstrap_data"
      value = "CHEF_POLICY_NAME=${local.chef["node_policy"]["disk"]}::CHEF_POLICY_GROUP=${local.chef["misc"]["policy_group"]}"
      propagate_at_launch = true
    },
    {
      key = "k8s.io/role/disk"
      value = "1"
      propagate_at_launch = true
    },
[...]
```

**Comment by phardy** [comment]
@postmaxin Use a [dynamic block](https://www.terraform.io/language/expressions/dynamic-blocks). For my use, we were generating tags based on a map like this

```
tags = [for k, v in var.tags: {
  key                 = k
  value               = v
  propagate_at_launch = false
}]
```

and have replaced it with this
```
dynamic "tag" {
  for_each = var.tags
  content {
    key                 = tag.key
    value               = tag.value
    propagate_at_launch = false
  }
}
```

**Comment by omerh** [comment]
Adding extra info for non static locals that might help

in addition to the static tags defined in a local


```
locals {
  tags = {
    Name = "eks-${var.name}"
    "${var.group}" = true
  }
}
```

and an external map variable added and merged

```
variable "tags" {
  type    = map(string)
  default = {}
}
```

I've added a count index, that must be inside the auto scale group as follow

```
dynamic "tag" {
    for_each = merge(local.cluster_tags, var.tags, {az_count = count.index})
    content {
      key = tag.key
      value = tag.value
      propagate_at_launch = true
    }
  }
```

Works on `4.12.1`
Hope it helps anyone

**Comment by jar-b** [comment]
Closed by #30842, merged to `main` via #31392
